### ISSUE TITLE
No progress when vapor build 

### ISSUE THREAD

**Issue by jerryga** [vapor/vapor#1946]
Run the vapor build command, no progress in the consle.
It have been stuck for 10 minutes.

[Xcode:10.1]

The Package.swift is as follows:

// swift-tools-version:4.0
import PackageDescription

let package = Package(
    name: "Hello",
    products: [
        .library(name: "Hello", targets: ["App"]),
    ],
    dependencies: [
        // 💧 A server-side Swift web framework.
        .package(url: "https://github.com/vapor/vapor.git", from: "3.0.0"),

        // 🔵 Swift ORM (queries, models, relations, etc) built on SQLite 3.
        .package(url: "https://github.com/vapor/fluent-sqlite.git", from: "3.0.0")
    ],
    targets: [
        .target(name: "App", dependencies: ["FluentSQLite", "Vapor"]),
        .target(name: "Run", dependencies: ["App"]),
        .testTarget(name: "AppTests", dependencies: ["App"])
    ]
)

**Comment by jerryga** [comment]
Run this command: swift package generate-xcodeproj
An Error occured: 
/usr/local/include/stdbool.h:4:10: error: #include nested too deeply
#include <stdbool.h>
         ^
<unknown>:0: error: could not build Objective-C module 'Darwin'

**Comment by 0xTim** [comment]
@jerryga that sounds like a borked Swift install - what does `swift --version` show?

**Comment by jerryga** [comment]
> @jerryga that sounds like a borked Swift install - what does `swift --version` show?

I just updated Xcode, the Xcode version number is: 10.2 The swift version number is:
 Apple Swift version 5.0 (swiftlang-1001.0.69.5 clang-1001.0.46.3)
Target: x86_64-apple-darwin18.5.0

**Comment by sayerbartlett** [comment]
I believe I am having a similar issue so I'll add my information here. I'm new to using Vapor and am attempting to build a test project. I am on a brand new Mac with a fresh install of brew, vapor, and Xcode. `vapor xcode` completes quickly with no errors, but does not actually create the *.xcodeproj file. 

**Swift Version**
```
swift --version
Apple Swift version 5.0 (swiftlang-1001.0.69.5 clang-1001.0.46.3)
Target: x86_64-apple-darwin18.2.0
```
**Xcode Version**
Version 10.2 (10E125)


**Terminal Output**
```
➜  Developer vapor new Test
Cloning Template [Done]
Updating Package Name [Done]
Initializing git repository [Done]
➜  Developer cd Test
➜  Test git:(master) vapor xcode
Generating Xcode Project [Done]
Select the `Run` scheme to run.
Open Xcode project?
y/n> y
Opening Xcode project...
Error: backgroundExecute(code: 1, error: "The file /Users/sbartlett/Developer/Test/*.xcodeproj does not exist.\n", output: "")
```


**Package.swift**
```
➜  Test git:(master) cat Package.swift
// swift-tools-version:4.0
import PackageDescription

let package = Package(
    name: "Test",
    products: [
        .library(name: "Test", targets: ["App"]),
    ],
    dependencies: [
        // 💧 A server-side Swift web framework.
        .package(url: "https://github.com/vapor/vapor.git", from: "3.0.0"),

        // 🔵 Swift ORM (queries, models, relations, etc) built on SQLite 3.
        .package(url: "https://github.com/vapor/fluent-sqlite.git", from: "3.0.0")
    ],
    targets: [
        .target(name: "App", dependencies: ["FluentSQLite", "Vapor"]),
        .target(name: "Run", dependencies: ["App"]),
        .testTarget(name: "AppTests", dependencies: ["App"])
    ]
)
```

**Comment by jerryga** [comment]
It is stucking here, NO any tips or progress
Generating Xcode Project [             •           ]

**Comment by 0xTim** [comment]
@jerryga can you run `sudo xcode-select -s /Applications/Xcode.app` and see if that fixes the issue?

@sayerbartlett does `vapor new HelloWorld` work?  (I'm wondering if Test is a reserved word or not) If not what does `swift package generate-xcodeproj` do?

**Comment by sayerbartlett** [comment]
Same results running `vapor new HelloWorld`

Running `swift package generate-xcodeproj`:
```
dyld: Library not loaded: @rpath/libswiftCore.dylib
  Referenced from: /Library/Developer/CommandLineTools/usr/bin/swift-package
  Reason: image not found
[1]    11728 abort      swift package generate-xcodeproj
```
Using that ^ output I was able to find the cause and solution mentioned [here](https://stackoverflow.com/questions/17980759/xcode-select-active-developer-directory-error).
Pointing `xcode-select` to the Xcode app Developer fixed it for me:
`sudo xcode-select -s /Applications/Xcode.app/Contents/Developer`

After that, `swift package generate-xcodeproj` worked. 
I tested running `vapor xcode` on a new Vapor project and it worked perfectly too 👌

I'm all set now - thanks for your help @0xTim

**Comment by jerryga** [comment]
> @jerryga can you run `sudo xcode-select -s /Applications/Xcode.app` and see if that fixes the issue?

Hi, It does not work.
`➜  ~ sudo xcode-select -s /Applications/Xcode.app
Password:
➜  ~ sudo xcode-select -s /Applications/Xcode.app
➜  ~ sudo xcode-select -s /Applications/Xcode.app
➜  ~ vapor new HelloWordVapor
Cloning Template [Done]
Updating Package Name [Done]
Initializing git repository [Done]
➜  ~ cd HelloWordVapor     
➜  HelloWordVapor git:(master) vapor xcode
Generating Xcode Project [ 
`

It’s stuck here.

**Comment by jerryga** [comment]
> Running `swift package generate-xcodeproj`:

When I run the `swift package generate-xcodeproj` command, the error occured:

`#include <sys/_types/_in_addr_t.h>
         ^

<unknown>:0: error: too many errors emitted, stopping now
<unknown>:0: error: could not build Objective-C module 'Darwin'
`

**Comment by vzsg** [comment]
Do you have MacPorts installed for any reason?

**Comment by jerryga** [comment]
> Do you have MacPorts installed for any reason?

NO. I did not install MacPorts. Should I install that?

**Comment by 0xTim** [comment]
No MacPorts screws things up. Is that the full output of `swift package generate-xcodeproj`?

**Comment by jerryga** [comment]
> No MacPorts screws things up. Is that the full output of `swift package generate-xcodeproj`?

`
➜  TestASA git:(master) swift package generate-xcodeproj
/Users/Asa/TestASA: error: manifest parse error(s):
<module-includes>:5:9: note: in file included from <module-includes>:5:
#import "copyfile.h"
        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/usr/include/copyfile.h:36:10: note: in file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/usr/include/copyfile.h:36:
#include <stdint.h>
         ^
/usr/local/include/stdint.h:59:11: note: in file included from /usr/local/include/stdint.h:59:
# include <stdint.h>
          ^
/usr/local/include/stdint.h:59:11: note: in file included from /usr/local/include/stdint.h:59:
# include <stdint.h>
          ^
/usr/local/include/stdint.h:59:11: note: in file included from /usr/local/include/stdint.h:59:
# include <stdint.h>
          ^
/usr/local/include/stdint.h:59:11: note: in file included from /usr/local/include/stdint.h:59:
# include <stdint.h>
          ^
/usr/local/include/stdint.h:59:11: note: in file included from /usr/local/include/stdint.h:59:
# include <stdint.h>
          ^
/usr/local/include/stdint.h:59:11: note: in file included from /usr/local/include/stdint.h:59:
# include <stdint.h>
          ^
/usr/local/include/stdint.h:59:11: note: in file included from /usr/local/include/stdint.h:59:
# include <stdint.h>
          ^
/usr/local/include/stdint.h:59:11: note: in file included from /usr/local/include/stdint.h:59:
# include <stdint.h>
          ^
/usr/local/include/stdint.h:59:11: note: in file included from /usr/local/include/stdint.h:59:
# include <stdint.h>
          ^
/usr/local/include/stdint.h:59:11: note: in file included from /usr/local/include/stdint.h:59:
# include <stdint.h>
          ^
/usr/local/include/stdint.h:59:11: note: in file included from /usr/local/include/stdint.h:59:
# include <stdint.h>
          ^
/usr/local/include/stdint.h:59:11: note: in file included from /usr/local/include/stdint.h:59:
# include <stdint.h>
          ^
/usr/local/include/stdint.h:59:11: note: in file included from /usr/local/include/stdint.h:59:
# include <stdint.h>
          ^
/usr/local/include/stdint.h:59:11: note: in file included from /usr/local/include/stdint.h:59:
# include <stdint.h>
          ^
………………………………

/usr/local/include/stdint.h:59:11: note: in file included from /usr/local/include/stdint.h:59:
# include <stdint.h>
          ^
/usr/local/include/stdint.h:72:11: note: in file included from /usr/local/include/stdint.h:72:
# include <sys/types.h>
          ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/usr/include/sys/types.h:110:10: error: #include nested too deeply
#include <sys/_types/_in_addr_t.h>
         ^
<unknown>:0: error: too many errors emitted, stopping now
<unknown>:0: error: could not build Objective-C module 'Darwin'

`
The log is too long, I omitted the similar part in the middle.
In addition, I have reinstalled the Xcode App, but it did not work for me.

**Comment by jerryga** [comment]
I have solved this problem by deleting the dir:/usr/local/include. By the way , thank you very much for the help of oxTim、tholo、gwynne and other team project memebers, thank you.

**Comment by Saifullahsajid** [comment]
I started learning vapor two days ago and my vapor build is stuck at `Updating https://github.com/vapor/core.git`. it took more than 20,30,40 mins and it won't proceed. This happened when I tried to add LEAF to vapor. Am I doing it wrong?

package.swift

```
// swift-tools-version:5.2
import PackageDescription

let package = Package(
    name: "FirstWebAppInVapor",
    platforms: [
       .macOS(.v10_15)
    ],
    dependencies: [
        // 💧 A server-side Swift web framework.
        .package(url: "https://github.com/vapor/vapor.git", from: "4.0.0"),
        .package(url: "https://github.com/vapor/fluent.git", from: "4.0.0"),
        .package(url: "https://github.com/vapor/leaf.git", from: "3.0.0"),
        .package(url: "https://github.com/vapor/fluent-mysql-driver.git", from: "4.0.0-rc")
    ],
    targets: [
        .target(
            name: "App",
            dependencies: [
                .product(name: "Fluent", package: "fluent"),
                .product(name: "FluentMySQLDriver", package: "fluent-mysql-driver"),
                .product(name: "Vapor", package: "vapor"),
		.product(name: "Leaf", package: "leaf")
            ],
            swiftSettings: [
                // Enable better optimizations when building in Release configuration. Despite the use of
                // the `.unsafeFlags` construct required by SwiftPM, this flag is recommended for Release
                // builds. See <https://github.com/swift-server/guides#building-for-production> for details.
                .unsafeFlags(["-cross-module-optimization"], .when(configuration: .release))
            ]
        ),
        .target(name: "Run", dependencies: [.target(name: "App")]),
        .testTarget(name: "AppTests", dependencies: [
            .target(name: "App"),
            .product(name: "XCTVapor", package: "vapor"),
        ])
    ]
)
```

**Comment by Saifullahsajid** [comment]
My issue got fixed by using leaf version `4.0.0-rc`
